### PR TITLE
add tftrue_autoupdate, and only check boot for update so as to not spam github's api

### LINKED ISSUE #29: [TFTrue] The update data hasn't been received. HTTP error 403. Response: {"message":"API rate limit exceeded for . (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
TFTrue shouldn't request an update so often. Maybe once on server start instead? Maybe add a cvar?

### PR DESCRIPTION
**Pull request by sapphonie**
fixes https://github.com/AnAkkk/TFTrue/issues/29

also cleans up spacing (to tabs) in that file so it's not mixed btween tabs/spaces

### CODE REVIEW

**Comment by sapphonie** [comment]
need to add a global (static?) bool to not run every map change

File: AutoUpdater.cpp [hunk 1]
```diff
@@ -75,221 +83,226 @@ bool CAutoUpdater::IsModuleValid(std::string strFileName)
...
-		{
-			tLastCheckTime = time(NULL);
+	g_AutoUpdater.PreCheckUpdate();
+}
 
-            CheckUpdate();
-		}
+// this is called after cfgs execute, so that,
+// if people specify tftrue_autoupdate 0 in their server cfg, tftrue won't ignore it
+void CMultiplayRules::LoadVoiceCommandScript()
```
Comment (AnAkkk) [inline AutoUpdater.cpp:93]: Hm, not sure I understand how this would be called in the plugin?

Comment (sapphonie) [reply AutoUpdater.cpp:93]: [`CMultiplayRules::CMultiplayRules()`](https://cs.alliedmods.net/hl2sdk-tf2/source/game/shared/multiplay_gamerules.cpp#220) [executes the `server.cfg` file](https://cs.alliedmods.net/hl2sdk-tf2/source/game/shared/multiplay_gamerules.cpp#234-247) and then [calls `LoadVoiceCommandScript` as its last func](https://cs.alliedmods.net/hl2sdk-tf2/source/game/shared/multiplay_gamerules.cpp#266), so LoadVoiceCommandScript is guaranteed to be called after server.cfg is executed. This allows server owners to set `tftrue_autoupdate` to 0 in their server.cfg without a race condition occuring (at least from what I can tell)

Comment (sapphonie) [reply AutoUpdater.cpp:93]: ^ Incorrect, there's no guarantees that the commands have been executed yet. Will probably have to implement a timer or something similar

Comment (AnAkkk) [reply AutoUpdater.cpp:93]: `CMultiplayRules::LoadVoiceCommandScript` is part of the game, and you don't seem to be hooking it, so it wouldn't be called at all in the plugin anyway?

Comment (sapphonie) [reply AutoUpdater.cpp:93]: I think I likely forgot to port the hooking code from my old pr but it's regardless not a usable way to handle this anyway as there's no way to guarantee that server.cfg has been executed. I'll have to think about this.

**Pull request closed without merging**